pkg/common/vclib/datacenter.go

```diff
@@ -439,3 +581,65 @@ func (dc *Datacenter) GetAllFirstClassDisks(ctx context.Context) ([]*FirstClassD
 
 	return firstClassDisks, nil
 }
+
+func (dc *Datacenter) DoesFirstClassDiskExist(ctx context.Context, fcdID string) (*FirstClassDiskInfo, error) {
+	datastores, err := dc.GetAllDatastores(ctx)
+	if err != nil {
+		glog.Errorf("GetAllDatastores failed. Err: %v", err)
+		return nil, err
+	}
+
+	for _, datastore := range datastores {
+		fcd, err := datastore.GetFirstClassDiskInfo(ctx, fcdID, FindFCDByID)
+		if err == nil {
+			glog.Infof("DoesFirstClassDiskExist(%s): FOUND", fcdID)
+			return fcd, nil
+		}
```

Comment:
May want to return `err` here if `err != ErrNoDiskIDFound` or at least log the `err`.

Reply:
I didn't want to flood the logs with errors. If you have 100 datastores and the FCD doesnt exist, you will have 100 lines in the log. Was just better to log it once.